### PR TITLE
Fix settings path being incorrect when file is not under repo root

### PR DESCRIPTION
**Pull request by madhurig**
Fix for issue reported in the docs repo: https://github.com/MicrosoftDocs/vsts-docs/issues/212#issuecomment-373290579

### CODE REVIEW

File: Tasks/Maven/maventask.ts [hunk 1]
```diff
@@ -181,7 +181,7 @@ async function execBuild() {
                             if ((options[i] === '--settings' || options[i] === '-s') && (i + 1) < options.length) {
                                 i++; // increment to the file name
                                 let suppliedSettingsXml: string = options[i];
-                                tl.cp(path.join(tl.cwd(), suppliedSettingsXml), settingsXmlFile, '');
```
Comment (jpricketMSFT) [inline Tasks/Maven/maventask.ts]: Can you explain how resolve is better than join here? Is it because suppliedSettingsXml might be a full path?

Comment (brcrista) [reply Tasks/Maven/maventask.ts]: You can just use `path.resolve(suppliedSettingsXml)`.  `resolve` will add the current working directory to create an absolute path if `suppliedSettingsXml` is relative, or return `suppliedSettingsXml` if it is absolute.

Comment (jpricketMSFT) [reply Tasks/Maven/maventask.ts]: tl.cwd() is not necessarily the current working directory.

Comment (madhurig) [reply Tasks/Maven/maventask.ts]: @jpricketMSFT: Yes, join doesn't work if the suppliedSettingsXml is a full path that is not under the cwd (see linked issue)
@brcrista : I think the way it is written is more readable :)

**Pull request merged**